app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script_source.rb

```diff
@@ -51,12 +73,23 @@ def queue(zone, instance_id, method_name, args, action)
   end
 
   def update_in_provider(params)
-    params.delete(:task_id) # in case this is being called through update_in_provider_queue which will stick in a :task_id
-    manager.with_provider_connection do |connection|
-      connection.api.projects.find(manager_ref).update_attributes!(params)
+    begin
+      params.delete(:task_id) # in case this is being called through update_in_provider_queue which will stick in a :task_id
+      manager.with_provider_connection do |connection|
+        connection.api.projects.find(manager_ref).update_attributes!(params)
+      end
+      self.class.send('refresh', manager)
+      reload
+    rescue AnsibleTowerClient::ClientError => tower_error
+      _log.error "update_in_provider failed: #{tower_error}"
+      error = tower_error
+    rescue ActiveRecord::RecordNotFound => not_found_error
+      _log.error "update_in_provider couldn't locate record: #{not_found_error}"
```

Comment:
I was thinking the case that the Tower object got deleted and refresh destroyed this.

But I guess chances are low. I've removed it.